include/deal.II/lac/block_vector_base.h

```diff
@@ -2264,6 +2262,17 @@ void BlockVectorBase<VectorType>::extract_subvector_to (ForwardIterator
     }
 }
 
+
+template <typename VectorType>
+inline
+bool BlockVectorBase<VectorType>::supports_distributed_data() const
+{
+  // Create a dummy vector of type VectorType to check
+  // supports_distributed_data even if BlockVectorBase contains no block
+  VectorType dummy;
+  return dummy.supports_distributed_data();
```

Comment:
Or, alternatively, if ```VectorSpaceVector``` (and similar) are meant to function as some sort of C++ concept to enforce a uniform interface - and we happen to not use the runtime polymorphism at all - then, we should use it strictly only on the template level. For example:
```
template<typename Vector2,
         typename = typename std::enable_if<std::is_base_of<VectorBaseVector<Number> Vector2<Number>>::value>::type>
Vector<number> &operator+=(const Vector2<Number> &V);
```
(This looks horrible, I know...)